File: tools/dbench/answers/go_haiku/s1/e1_parse_fallback.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	vals := []string{"12", "x", "7", "-", "30"}
	sum := 0
	unparseable := 0

	for _, v := range vals {
		n, err := strconv.Atoi(v)
		if err != nil {
			unparseable++
		} else {
			sum += n
		}
	}

	fmt.Println(sum)
	fmt.Println(unparseable)
}

File: tools/dbench/answers/rust_sonnet/s1/e1_parse_fallback.rs
fn main() {
    let values = ["12", "x", "7", "-", "30"];
    let mut sum = 0i64;
    let mut unparseable = 0usize;
    for s in &values {
        match s.parse::<i64>() {
            Ok(n) => sum += n,
            Err(_) => unparseable += 1,
        }
    }
    println!("{}", sum);
    println!("{}", unparseable);
}
